### ISSUE TITLE
Implement missing IO methods

### ISSUE THREAD

**Issue by postmodern** [postmodern/fake_io.rb#2]
Since ruby 1.9.1 some additional IO methods were added:

* [x] [pread](https://rubydoc.info/stdlib/core/IO#pread-instance_method)
* [x] [pwrite](https://rubydoc.info/stdlib/core/IO#pwrite-instance_method)
* [x] [advise](https://rubydoc.info/stdlib/core/IO#advise-instance_method)
* [x] [external_encoding](https://rubydoc.info/stdlib/core/IO#external_encoding-instance_method)
* [x] [internal_encoding](https://rubydoc.info/stdlib/core/IO#internal_encoding-instance_method)
* [x] [set_encoding_by_bom](https://rubydoc.info/stdlib/core/IO#set_encoding_by_bom-instance_method)
* [x] [set_encoding](https://rubydoc.info/stdlib/core/IO#set_encoding-instance_method)

**Comment by postmodern** [comment]
If we implement external/internal encoding switching, if `binmode?` is `true` then written Strings need to be converted to `ASCII-8BIT`.

**Comment by postmodern** [comment]
It appears under Ruby 2.7 that `#lines`, `#bytes`, `#chars`, `#codepoints` are indeed IO methods.

Fixed in fe897ee26aa318cc8a3a2c9e2b222bb1aa5b3f55.

**Comment by postmodern** [comment]
This is how I'm diffing the IO methods vs FakeIO's methods:

```ruby
#!/usr/bin/env ruby

$LOAD_PATH.unshift(File.expand_path(File.join(__dir__,'lib')))
require 'fake_io'

class TestIO
  include FakeIO
end

fake_io = TestIO.new

puts "Missing methods not defined in FakeIO:"
p IO.public_instance_methods - fake_io.public_methods

puts "Additional methods defined not in IO:"
p fake_io.public_methods - IO.public_instance_methods
```
